cadc-util/src/main/java/ca/nrc/cadc/net/AuthChallenge.java

```diff
@@ -0,0 +1,145 @@
+*  products derived from this           promouvoir les produits dérivés
+*  software without specific prior      de ce logiciel sans autorisation
+*  written permission.                  préalable et particulière
+*                                       par écrit.
+*
+*  This file is part of the             Ce fichier fait partie du projet
+*  OpenCADC project.                    OpenCADC.
+*
+*  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+*  you can redistribute it and/or       vous pouvez le redistribuer ou le
+*  modify it under the terms of         modifier suivant les termes de
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package ca.nrc.cadc.net;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.apache.log4j.Logger;
+
+/**
+ * Class to capture the value of a www-authenticate challenge.
+ * 
+ * @author pdowler
+ */
+public class AuthChallenge {
+    private static final Logger log = Logger.getLogger(AuthChallenge.class);
+
+    private final String name;
+    private Map<String,String> params = new TreeMap<>();
+        
+    /**
+     * Parse a www-authenticate challenge of the form
+     * name [key="value"[, key="value" ...]]
+     * 
+     */
+    public AuthChallenge(String challenge) {
+        String s = challenge.replace(",", " "); // remove comma(s) between k=v pairs
+
+        String[] parts = s.split("[ \t]");
+        this.name = parts[0];
+        if (parts.length == 1) {
```

Comment:
can parts.length be < 1?

Reply:
if the input string is not null, there is at least one part.

if it is null: NPE from the first line of the ctor, which I'm ok with at this point